docs/guides/tfhers.md

```diff
@@ -0,0 +1,242 @@
+# TFHE-rs Compatibility
+
+This document explains how to combine Concrete and [TFHE-rs](https://github.com/zama-ai/tfhe-rs) computations together. This feature targets two main use cases.
+
+## Table of Content
+
+- [The Basics](#the-basics)
+- [First Use Case](#first-use-case-shared-secret-key)
+- [Second Use Case](#second-use-case-pregenerated-tfhe-rs-keys)
+- [Serialization of Ciphertexts and Keys](#serialization-of-ciphertexts-and-keys)
+
+## The Basics
+
+There are differences between Concrete and TFHE-rs, so ensuring compatibility between them involves more than just data serialization. To achieve compatibility, we need to consider two main aspects.
+
+1. In TFHE-rs and Concrete, integers are encoded differently. Both use LWE ciphertexts, but they organize them differently to represent integers. In Concrete, TFHE-rs big integers are represented as a vector of `n` Concrete integers. `n` is determined by dividing the TFHE-rs integer bit-width by the TFHE-rs message bit-width. These bit-width values, along with the carry bit-width, are the encoding parameters used in Concrete to represent TFHE-rs big integers. When working with a TFHE-rs integer type in Concrete, you can use the `.encode(...)` and `.decode(...)` functions to see this in practice.
+
+2. The Concrete Optimizer doesn't limit its search space to available TFHE-rs parameters. Therefore, we need to either fix or limit the search space in areas of the circuit where compatibility is required. This will ensure that the same parameters are used where needed.
+
+## First Use Case: Shared Secret Key
+
+The first scenario involves a single party seeking to combine both Concrete and TFHE-rs in a computation. In this case, a shared secret key will be used, while different keysets will be held for Concrete and TFHE-rs. It is possible to generate keys in both directions, meaning that you can start by generating keys from Concrete and then use Concrete's secret key to generate the remaining keys (partial keygen) in TFHE-rs (use case 1.1), or vice versa (use case 1.2).
+
+After the key generation is complete and both keysets are acquired, we can perform computations, encryption, and decryption on both ends.
+
+![alt text](../_static/tfhers/tfhers_uc1.svg)
+
+### Initial Setup and Config
+
+The first step is to define the TFHE-rs ciphertext we want to use (see [The Basics](#the-basics)). This includes specifying cryptographic and encoding parameters.
+
+```python
+from functools import partial
+from concrete.fhe import tfhers
+
+tfhers_params = tfhers.CryptoParams(
```

Comment:
explain where it comes from, or, if the user doesn't have to change that, say it